src/util/fs.ts

```diff
@@ -172,5 +175,77 @@ export const fs = {
     } catch (err) {
       return false;
     }
+  },
+
+  /**
+   * Recursively act on all files or directories in a directory
+   *
+   * @param dir path to directory
+   * @param perform function to be run on contents of dir
+   * @param onType optional parameter to specify type to actOn
+   * @returns void
+   */
+
+  actOn: async (dir: string, perform: PerformFunction, onType: 'file' | 'dir' = 'file'): Promise<void> => {
+    for (const file of await fs.readdir(dir)) {
+      const filePath = path.join(dir, file);
+      const stat = await fs.stat(filePath);
+
+      if (stat) {
+        if (stat.isDirectory()) {
+          await fs.actOn(filePath, perform, onType);
+          if (onType === 'dir') {
+            await perform(filePath);
+          }
+        } else if (stat.isFile() && onType === 'file') {
+          await perform(filePath, file, dir);
+        }
+      }
+    }
+  },
+
+  /**
+   * Checks if files are the same
+   *
+   * @param file1Path the first file path to check
+   * @param file2Path the second file path to check
+   * @returns boolean
+   */
+  areFilesEqual: async (file1Path: string, file2Path: string): Promise<boolean> => {
+    try {
+      const file1Size = (await fs.stat(file1Path)).size;
+      const file2Size = (await fs.stat(file2Path)).size;
+      if (file1Size !== file2Size) {
+        return false;
+      }
+    } catch (err) {
+      throw new SfdxError(`no such file or directory, stat ${err.path}`, 'DirMissingOrNoAccess');
```

Comment:
Why throw a different error message? You can really have just one try-catch. You can also have the last return in the try as well so you don't have to have the contentA/contentB outside of the try.